### ISSUE TITLE
Where to find the dataset used for training pretrained unet model?

### ISSUE THREAD

**Issue by SachidanandAlle** [mostafajahanifar/nuclick_torch#12]
Like to know https://drive.google.com/file/d/1d_ypVYTsXoMrTVJaEfVRGS5CfLkxyViK/view?usp=sharing is trained on which dataset?

Is it https://warwick.ac.uk/fac/cross_fac/tia/data/nuclick/?
Under this there are two.. both were used to train the above model.. or one of them?

1 - Images of white blood cells (WBC) in blood sample images with their segmentation masks ([downloadLink opens in a new window](https://warwick.ac.uk/fac/cross_fac/tia/data/nuclick/hemato_data.zip))

2 - Datasets of lymphocyte segmentation in Immunohistochemistry (IHC) images ([download](https://warwick.ac.uk/fac/cross_fac/tia/data/nuclick/ihc_nuclick.zip))

**Comment by SachidanandAlle** [comment]
After training 100 epochs on WBC samples from nuclick datasets, I can pretty confirm that the UNET model (pretrained) shared through g-drive is not just trained `Images of white blood cells (WBC)`..  

Will be helpful if you can share all the configs (if something different than default config in the code) and datasets that were used to train the model..

**Comment by SachidanandAlle** [comment]
I see your pretrained model is mostly trained on PanNuke dataset... or combination of all..

**Comment by mostafajahanifar** [comment]
Dear @SachidanandAlle,

Thanks for your interest and raising this issue. As you figured out, the provided pretrain models are trained on a combination of PanNuke and Lizard datasets which are nuclei segmentation dataset. To make this clear, I should say that the `White Blood Cells` and `Lymphocyte` datasets are auxiliary products of NuClick paper where the first one is a cell (not nuclei) segmentation dataset and the second one is actually NuClick's output. It means that you should not expect a model trained on WBCs to work good on nuclei images. Anyway, the current provided models have not been trained on any of these two datasets.

However, we are planning to train NuClick separately on  different datasets and report the results and provide the weights separately. Hopefully, that would help.

p.s.
We used the default config in the code for training of the networks. However, we will mention in any changes in the config when we are releasing the weights for different datasets.

**Comment by SachidanandAlle** [comment]
Thanks for the answers... This is basically to integrate nuclick model in MONAILabel.
https://github.com/Project-MONAI/MONAILabel/pull/772#issue-1228981474

I could add NuClick for both train and infer in MONAILabel.  However I had to use MONAI network/components.. and for generating signals, I am not using distance cv2.distanceTransform to find the random points for inclusion map based on distance.. currently using simple random for 128x128 patch.. also some basic pre-transforms from torch vision via MONAI.. (hence i might be ignoring some model performance improvements that you might have in your pipeline;  but objective is not to do exact match with metrics that you have in your pipeline.. i will let users to finetune such things for themselves)

With this, in MONAILabel can browse any wsi image, add some annotations over new cells and retrain nuclick model for the new data/cells.

That's one reason I was looking for what dataset you had used to produce the pretrained models.  For now, I have used only PanNuke and I guess it's a good start for MONAILabel users.

Based on the above details, let me know if you have any comments.  I guess metrics are kind of similar to what you might have got.  I could run some manual checks on some nuclei cells.  I felt ok with the boundaries generated by the new model.

May be if you are interested, you can give a quick try in QuPath using MONAILabel plugin+server..

**Comment by SachidanandAlle** [comment]
Not using random sampling.. and yes only picking a random point inside of labeled nuclei (for inclusion map)
https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/pathology/lib/trainers/nuclick.py#L276-L279

Yes.. you should give a try.. we can make qupath click interaction better.. but didn't invest too much time.. just used existing annotation tools to add pos point and then run.. but someone can make it better.. click and auto run (similar to ohif/3d slicer for deepgrow).. more of frontend/ui work..
